sdcm/tester.py

```diff
@@ -2567,11 +2569,16 @@ def collect_partitions_info(self, table_name, primary_key_column, save_into_file
                 try:
                     with self.db_cluster.cql_connection_patient(node=self.db_cluster.nodes[0],
                                                                 connect_timeout=600) as session:
-                        session.default_consistency_level = ConsistencyLevel.QUORUM
-                        result = session.execute(count_pk_rows_cmd)
-                        pk_rows_num_result = result.current_rows[0].count
+                        result = session.execute_async(SimpleStatement(
```

Comment:
there is already a `fetch_all_rows()`.. isn't it better to use it, instead of all what you have done here?

Reply:
fetch_all_rows() is a nice option. In this case it will only replace 2 lines with 1 line so not too much gain.
The problems why not using `fetch_all_rows()` are:
The retrying of `n=4, sleep_time=5` is not not needed here and can waste huge amount of test time.
`PageFetcher(result).request_all()` is missing `timeout` parameter.

Reply:
I don't understand where's the waste of time.

And what is the problem with the retry?

Anyhow if you think you need the retries as part of the function parameters, refactor the function so it can accommodate your needs as well.

It's replacing more than one line of code, and it's doing exactly what your code is doing, but with a retry.

Reply:
fixed.